dart/dynamics/BodyNode.cpp

```diff
@@ -888,34 +960,16 @@ void BodyNode::init(Skeleton* _skeleton)
 }
 
 //==============================================================================
-void BodyNode::draw(renderer::RenderInterface* _ri,
-                    const Eigen::Vector4d& _color,
-                    bool _useDefaultColor,
-                    int _depth) const
+void BodyNode::processNewEntity(Entity* _newChildEntity)
 {
-  if (_ri == NULL)
+  if(find(mChildBodyNodes.begin(), mChildBodyNodes.end(), _newChildEntity) !=
+     mChildBodyNodes.end())
```

Comment:
The purpose of this function is probably not obvious, so I'll add an explanation in the comments of the code. The point of this function is to sort out the arbitrary Entities that get added from the BodyNode children that are getting added.

This might sound silly, but there are some performance optimizations that can be achieved in (for example) the update notifications by sorting out the BodyNode child Entities from the other arbitrary types of child Entities. It has to do with compile-time resolution of virtual functions. If I call notifyTransformUpdate() on an arbitrary Entity pointer, a runtime vtable lookup will be necessary, which means the compiler can't do any inline optimizations. On the other hand, if I call notifyTransformUpdate() on a BodyNode pointer, it can resolve the vtable lookup at compile time, which allows the compiler to inline.